bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/ApiBridgeHandler.java

```diff
@@ -89,138 +91,150 @@
 
+    private @Nullable OAuthClientService oAuthClientService;
     private Optional<ScheduledFuture<?>> connectJob = Optional.empty();
     private Optional<WebhookServlet> webHookServlet = Optional.empty();
     private Optional<GrantServlet> grantServlet = Optional.empty();
 
     public ApiBridgeHandler(Bridge bridge, HttpClient httpClient, NADeserializer deserializer,
-            BindingConfiguration configuration, HttpService httpService) {
+            BindingConfiguration configuration, HttpService httpService, OAuthFactory oAuthFactory) {
         super(bridge);
         this.bindingConf = configuration;
         this.httpClient = httpClient;
         this.deserializer = deserializer;
         this.httpService = httpService;
-        this.requestCountChannelUID = new ChannelUID(thing.getUID(), GROUP_MONITORING, CHANNEL_REQUEST_COUNT);
+        this.oAuthFactory = oAuthFactory;
 
-        Path homeFolder = Paths.get(OpenHAB.getUserDataFolder(), BINDING_ID);
-        if (Files.notExists(homeFolder)) {
-            try {
-                Files.createDirectory(homeFolder);
-            } catch (IOException e) {
-                logger.warn("Unable to create {} folder : {}", homeFolder.toString(), e.getMessage());
-            }
-        }
-        tokenFile = homeFolder.resolve(REFRESH_TOKEN + "_" + thing.getUID().toString().replace(":", "_"));
+        requestCountChannelUID = new ChannelUID(thing.getUID(), GROUP_MONITORING, CHANNEL_REQUEST_COUNT);
     }
 
     @Override
     public void initialize() {
         logger.debug("Initializing Netatmo API bridge handler.");
+
+        ApiHandlerConfiguration configuration = getConfiguration();
+
+        if (configuration.clientId.isBlank()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    ConfigurationLevel.EMPTY_CLIENT_ID.message);
+            return;
+        }
+
+        if (configuration.clientSecret.isBlank()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    ConfigurationLevel.EMPTY_CLIENT_SECRET.message);
+            return;
+        }
+
+        oAuthClientService = oAuthFactory
+                .createOAuthClientService(this.getThing().getUID().getAsString(),
+                        AuthenticationApi.TOKEN_URI.toString(), AuthenticationApi.AUTH_URI.toString(),
+                        configuration.clientId, configuration.clientSecret, FeatureArea.ALL_SCOPES, false)
+                .withGsonBuilder(new GsonBuilder().registerTypeAdapter(AccessTokenResponse.class,
+                        new AccessTokenResponseDeserializer()));
+
         updateStatus(ThingStatus.UNKNOWN);
+
         scheduler.execute(() -> openConnection(null, null));
     }
 
     public void openConnection(@Nullable String code, @Nullable String redirectUri) {
-        ApiHandlerConfiguration configuration = getConfiguration();
+        authenticate(code, redirectUri);
```

Comment:
Something that might be strange. You are not considering the result of authenticate and just after you start the webhook servlet and set the thing status to ONLINE.
But if `authenticate` leads to the call of `startAuthorizationFlow`, I have the feeling you should not do what is just after this line ?

Reply:
I agree this can be refactored further.

I believe the current logic corresponds to the previous behavior, so as a consequence of keeping the scope to replacing the OAuth2 implementation, this has not yet been addressed.

Previously we would read refresh token from file, and if not blank, the logic from line 164 (`case COMPLETED`) in the old code on the left side would be triggered.

I already had some additional refactoring on my TODO for a follow-up PR, but I can also enhance it further in this PR.

Reply:
I just do not understand how we can be in status configuration error with the current code. 
Strangely I saw this status before authenticating. 
There is certainly something obvious I do not see because my feeling is that the current code should always lead to an ONLINE status... except if there is an uncaught exception preventing to reach the end of this method.

Reply:
You are right:
```
2023-04-21 15:14:58.004 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'netatmo:account:b61fd88486' changed from UNINITIALIZED (DISABLED) to INITIALIZING
2023-04-21 15:14:58.013 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'netatmo:account:b61fd88486' changed from INITIALIZING to UNKNOWN
2023-04-21 15:14:58.188 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'netatmo:account:b61fd88486' changed from UNKNOWN to OFFLINE (CONFIGURATION_ERROR): Configuration incomplete, please grant the binding to Netatmo Connect.
2023-04-21 15:14:58.188 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'netatmo:account:b61fd88486' changed from OFFLINE (CONFIGURATION_ERROR): Configuration incomplete, please grant the binding to Netatmo Connect. to ONLINE
```